Shared/Sources/FocusConfiguration.cpp

```diff
@@ -0,0 +1,76 @@
+//
+// Created by Etienne Pasteur on 27/03/2018.
+//
+
+#include <iostream>
+#include <fstream>
+#include <spdlog/spdlog.h>
+#include "FocusConfiguration.hpp"
+#include "lightconf/lightconf.hpp"
+#include "lightconf/config_format.hpp"
+
+namespace lightconf {
+    LIGHTCONF_BEGIN_ENUM(serverType)
+        LIGHTCONF_ENUM_VALUE(serverType::AUTHENTICATION, "AUTHENTICATION")
+        LIGHTCONF_ENUM_VALUE(serverType::BACKEND, "BACKEND")
+    LIGHTCONF_END_ENUM()
+    LIGHTCONF_BEGIN_TYPE(user)
+        LIGHTCONF_TYPE_MEMBER_OPT(std::string, _first_name, "first_name", "foo")
+        LIGHTCONF_TYPE_MEMBER_OPT(std::string, _last_name, "last_name", "bar")
+        LIGHTCONF_TYPE_MEMBER_REQ(std::string, _email, "email")
+        LIGHTCONF_TYPE_MEMBER_REQ(std::string, _password, "password")
+    LIGHTCONF_END_TYPE()
+
+    LIGHTCONF_BEGIN_TYPE(server)
+        LIGHTCONF_TYPE_MEMBER_REQ(std::string, _ip, "ip")
+        LIGHTCONF_TYPE_MEMBER_REQ(int, _port, "port")
+        LIGHTCONF_TYPE_MEMBER_REQ(serverType, _type, "type")
+    LIGHTCONF_END_TYPE()
+}
+
+FocusConfiguration::FocusConfiguration(const std::string &configFile) {
+    lightconf::group config_group;
+
+    std::string source;
+    try {
+        std::ifstream stream(configFile, std::ios::in);
+        if (stream) {
+            source = std::string(std::istreambuf_iterator<char>(stream),
+                                 std::istreambuf_iterator<char>());
+            config_group = lightconf::config_format::read(source);
+            _userInfo = config_group.get<user>("user");
+            _serversInfo = config_group.get<std::vector<server>>("servers", {});
+            _filed = true;
+        } else {
+            _filed = false;
+            spdlog::get("logger")->error("Missing configuration file");
+        }
+    } catch (const lightconf::parse_error &e) {
+        _filed = false;
+        spdlog::get("logger")->error("Parsing configuration file error");
+    } catch (const std::exception &e) {
+        _filed = false;
+        spdlog::get("logger")->error("Failed to read configuration file");
+    }
+}
+
+bool FocusConfiguration::isFiled() const{
+    return _filed;
+}
+
+struct user FocusConfiguration::getUser() const{
+    return _userInfo;
+}
+
+struct server FocusConfiguration::getServer(const serverType type) const{
```

Comment:
getters shouldn't have the side effect of creating a default server. This belongs in the constructor or param fetching